lib/cluster/index.ts

```diff
@@ -890,6 +956,49 @@ class Cluster extends EventEmitter {
     });
   }
 
+  private resolveSrv(hostname: string): Promise<any> {
+    return new Promise((resolve, reject) => {
+      dns.resolveSrv(hostname, function (err, addrs) {
+        if (err) {
+          /* no SRV record, try domain as A */
+          resolve(err);
+        } else {
+          let pending = 0,
+            error,
+            results = [];
+          let cb1 = function (e, addrs1) {
+            error = error || e;
+            results = results.concat(addrs1);
+            pending--;
+            if (pending < 1) {
+              if (results != null) {
+                resolve(results);
+              } else {
+                reject(error);
+              }
+            }
+          };
+          // cb1(null,addrs[0])
+          let gSRV = groupSrvRecords(addrs);
+          pending = gSRV.length;
+          gSRV.forEach(function (addr) {
+            resolveHost(addr.name, function (e, a) {
```

Comment:
I'd suggest `resolveHost` returns a Promise so we can use `Promise.all` to avoid nested callbacks.